### ISSUE TITLE
Default/Fallback Values

### ISSUE THREAD

**Issue by MichaelHirn** [jamesramsay/hercule#40]
We are using hercule for generating our finished API Blueprint. By now a lot of thought and try/error went into *how to best create DRY blueprints?* and one thing that turned out to be very helpful were *default* values. So we have this atomic MSON objects, from whereon we generate (thanks to hercule) more complex MSON objects (which keeps everything super DRY). Now the problem is, that an MSON object might have slightly different settings based on where it gets used e.g. response vs. request (which makes the approach only super DRY, but not as DRY and flexible as possible). A quick example.

``` 
[file: title.md]
- title: None (string, required)

    The title
```

So in our response this atomic MSON object would be quite alright, but in our request we face the problem, that it might not be `required`, therefore we would have to repeat ourselves all over again. But support for default values could solve the problem.

```
[file: title.md]
- title: None (string, :[Presence Placeholder](presence || "required"))

    The title
```

Now we can reuse this atomic MSON object wherever we need it.

```
## Get Product Response Attribute

:[Product Title](title.md)

## Create Product Request Attribute

:[Product Title](title.md presence:"optional")
```

This also comes in handy for different sample values for different requests/responses and testing the blueprint with Dredd.

Now, if that is too abstract and a bigger picture is missing you can check out our [example API Blueprint](https://github.com/storeness/api-blueprint/tree/master/examples) that documents and outlines how we use `hercule` to create DRY API Blueprints.

Also, I already wrote the code for providing support for default values. You can find the [fork here](https://github.com/MichaelHirn/hercule).

I am not sure though if that is something that proves valuable to you, yet. Maybe you already have some thoughts about the support for default/fallback values? I am just curious what you think about it and if you could imagine to integrate something like this into the project.

**Comment by jamesramsay** [comment]
@MichaelHirn Cool idea! I'm currently we are doing a similar thing:

```
# Compulsory
:[Example transclude](test.md presence:)

# Optional
:[Example transclude](test.md presence:"optional")
```

This works for us because, using your example, we state up front that all attributes are `required` unless other specified. But I can definitely see your use case!

**Comment by jamesramsay** [comment]
Merged and deployed to NPM in v1.2.0
